test/index.spec.js

```diff
@@ -0,0 +1,78 @@
+const debug = require('debug');
+const express = require('express');
+const assert = require('node:assert');
+const { spawn } = require('node:child_process');
+const { after, before, beforeEach, describe, it } = require('node:test');
+const { resolve } = require('path');
+
+const log = debug('hck').extend('test');
+
+describe('@hackolade/fetch library', () => {
+  let server;
+  const initiators = [];
+
+  function startServer() {
+    log('starting server');
+    const app = express();
+    const ok = (initiator) => (req, res) => {
+      log('received request from %o', initiator);
+      initiators.push(initiator);
+      res.status(200).end();
+    };
+    app.get('/main', ok('main'));
+    app.get('/renderer', ok('renderer'));
+    app.get('/utility', ok('utility'));
+    server = app.listen(3000);
+    log('server is listening');
+  }
+
+  function stopServer() {
+    log('stopping server');
+    server.close();
+  }
+
+  function clearInitiators() {
+    initiators.splice(0, initiators.length);
+  }
+
+  function wait(ms) {
+    return new Promise((resolve) => {
+      setTimeout(resolve, ms);
+    });
+  }
+
+  async function startElectronApp() {
+    const appProcess = spawn('npm', ['run', 'test:app:electron'], {
+      cwd: resolve(__dirname, '..'),
+    });
+    appProcess.stdout.on('data', (data) => log('STDOUT: %o', data.toString()));
+    appProcess.stderr.on('data', (data) => log('STDERR: %o', data.toString()));
+    await wait(2000);
+    appProcess.kill();
+  }
+
+  function assertServerReachedFrom(initiator) {
+    assert.ok(initiators.includes(initiator));
+  }
+
+  before(() => startServer());
+
+  beforeEach(() => clearInitiators());
+
+  after(() => stopServer());
+
+  it('should reach the server from the main process through a direct connection', async () => {
+    await startElectronApp();
+    assertServerReachedFrom('main');
```

Comment:
can we minimize the number of hardcoded strings? I assume that main/renderer/utitlity are involved and actually **depends** on the routes, as well as should match with the one in [resources/electron-app](https://github.com/hackolade/fetch/pull/1/files#diff-1b28fd592267a952cb167feda16da923cb033ba2d6ca6782f7474c6091921c57R15) folder

Reply:
I tried extracting the 3 constants to a separate file but it is not that simple because I have a mix of CommonJS (main) and ESM (renderer). So if I make it work for one process, it breaks the other and vice-versa. No worth the time in this case I think...